sds_data_manager/stacks/dynamodb_stack.py

```diff
@@ -0,0 +1,139 @@
+
+            The combination of the partition key and sort key together must be unique for each item
+            in a DynamoDB table. This means that while multiple items within a partition can have the
+            same sort key value, their partition key values must be different.
+        env : Environment
+            Account and region
+        on_demand : bool
+            If true, creates on demand DynamoDb table. If false, creates provisioned DynamoDb table.
+        read_capacity : int
+            Read capacity for provisioned DynamoDb table.
+            Default value is 1.
+        write_capacity : int
+            Write capacity for provisioned DynamoDb table.
+            Default value is 1.
+        """
+        self.sds_id = sds_id
+        self.table_name = table_name
+        self.partition_key = partition_key
+        self.sort_key = sort_key
+        self.on_demand = on_demand
+        if self.on_demand:
+            self._create_on_demand_dynamodb()
+        else:
+            self._create_provisioned_dynamodb(read_capacity, write_capacity)
+
+    def _create_on_demand_dynamodb(self):
+        """Creates On Demand DynamoDb table. On Demand DynamoDb table is created with PAY_PER_REQUEST billing mode.
+
+        When you turn on point-in-time recovery (PITR), DynamoDB backs up your table data automatically so that you
+        can restore to any given second in the preceding 35 days.
+        Returns
+        -------
+        Construct
+            DynamoDb table construct
+        """
+        return dynamodb.Table(
+            self,
+            f"OnDemandDynamoDb-{self.sds_id}",
+            table_name=self.table_name,
+            partition_key=dynamodb.Attribute(
+                name=self.partition_key, type=dynamodb.AttributeType.STRING
+            ),
+            sort_key=dynamodb.Attribute(
+                name=self.sort_key, type=dynamodb.AttributeType.STRING
+            ),
+            billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
+            removal_policy=RemovalPolicy.DESTROY,
+            point_in_time_recovery=True,
+        )
+
+    def _create_provisioned_dynamodb(self, read_capacity: int, write_capacity: int):
+        """Creates Provisioned DynamoDb table. Provisioned DynamoDb table is created with PROVISIONED billing mode.
+
+        When you turn on point-in-time recovery (PITR), DynamoDB backs up your table data automatically so that you
+        can restore to any given second in the preceding 35 days.
+
+        Parameters
+        ----------
+        read_capacity : int
+            Read capacity for provisioned DynamoDb table.
+        write_capacity : int
+            Write capacity for provisioned DynamoDb table.
+
+        Returns
+        -------
+        Construct
+            DynamoDb table construct
+        """
+        if read_capacity == 1 and write_capacity == 1:
+            raise Warning(
```

Comment:
Ah ok. I didn't read carefully before. I just saw this `Can only be provided if billingMode is Provisioned.` and thought I need to set those two to some value when provisioned. But it looks like CDK defaults to 5 which is good but not idea. I feel like it might be give a warning of some sort if someone wants to create provisioned version, with an assumption that user may be needing higher read and write capacity than 5 if they chose to provisioned resources.